### PR TITLE
[5422] AttachmentsPickerSystemTabFactory configuration and bug fixes

### PR DESCRIPTION
**Pull request by VelikovPetar**
### 🎯 Goal

Fixes couple of UI bugs in the new `AttachmentsPickerSystemTabFactory` and makes its content configurable.
Resolves: https://github.com/GetStream/stream-chat-android/issues/5422

### 🛠 Implementation details

1. Make `AttachmentsPickerSystemTabFactory` configurable: It now accepts a list of the following flags, instructing which types of attachment pickers it supports:
- `filesAllowed` - if the file attachments picker is allowed
- `mediaAllowed` - if the photo attachment picker is allowed
- `captureImageAllowed` - if the photo capture option is allowed
- `captureVideoAllowed` - if the video capture option is allowed
- `pollAllowed` - if the poll option is allowed
The `AttachmentsPickerTabFactories#defaultFactoriesWithoutStoragePermissions` is also extended with the same options, similar to the already existing `AttachmentsPickerTabFactories#defaultFactories`
2. Fix for the case where on smaller screens, the picker options were 'squashed' - now the list of options is scrollable, so if there is not enough room for all options, the user will be able to scroll to see all options.
3. Fix for the edge case where we must request the `CAMERA` permission, even when using the `AttachmentsPickerSystemTabFactory`: If the `CAMERA` permission is declared in the app manifest and is not granted, we still must request the permission, before staring the camera capture. (if the `CAMERA` permission is not declared in the Manifest, or is declared and granted, it works properly)
4. Additionally, align the picker buttons size and spacing in the XML SDK to the Compose SDK.

### 🎨 UI Changes

_Add relevant videos_

<table>
<thead>
<tr>
<th>SDK</th>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
XML
</td>
<td>
<video src="https://github.com/user-attachments/assets/1f053e55-b8bf-4e7a-80e7-b26b6d216e79
" controls="controls" muted="muted" />
</td>
<td>
<video src="https://github.com/user-attachments/assets/aec16689-9259-4c11-9e7f-36e90cc6fcb2
" controls="controls" muted="muted" />
</td>
</tr>
<tr>
<td>
Compose
</td>
<td>
<video src="https://github.com/user-attachments/assets/d41aa8bb-65c0-44a4-a504-c5e1068b4696
" controls="controls" muted="muted" />
</td>
<td>
<video src="https://github.com/user-attachments/assets/fb7885e6-80d5-4d4a-81d1-3785f789ccd1" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

Permissions bug:
1. Ensure the system media picker factory is used (`useDefaultSystemMediaPicker`)
2. Ensure the `CAMERA` permission is declared in the Manifest, but is not granted
3. Open a chat
5. Tap on the add attachments button
6. Tap on "Capture"

<details>

<summary>Patch for permissions bug testing</summary>

```
Subject: [PATCH] System media picker testing.
---
Index: stream-chat-android-ui-components-sample/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/AndroidManifest.xml b/stream-chat-android-ui-components-sample/src/main/AndroidManifest.xml
--- a/stream-chat-android-ui-components-sample/src/main/AndroidManifest.xml	
+++ b/stream-chat-android-ui-components-sample/src/main/AndroidManifest.xml	
@@ -19,6 +19,7 @@
     >
 
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.CAMERA" />
 
     <application
         android:name="io.getstream.chat.ui.sample.application.App"
Index: stream-chat-android-compose-sample/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/AndroidManifest.xml b/stream-chat-android-compose-sample/src/main/AndroidManifest.xml
--- a/stream-chat-android-compose-sample/src/main/AndroidManifest.xml	
+++ b/stream-chat-android-compose-sample/src/main/AndroidManifest.xml	
@@ -19,6 +19,8 @@
     package="io.getstream.chat.android.compose.sample"
     >
 
+    <uses-permission android:name="android.permission.CAMERA" />
+
     <application
         android:name="io.getstream.chat.android.compose.sample.ChatApp"
         android:allowBackup="false"
Index: stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml b/stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml
--- a/stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml	
+++ b/stream-chat-android-ui-components-sample/src/main/res/layout/fragment_chat.xml	
@@ -52,6 +52,7 @@
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/messageListView"
+        app:streamUiMessageComposerAttachmentsPickerSystemPickerEnabled="true"
         />
 
 </androidx.constraintlayout.widget.ConstraintLayout>
Index: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt
--- a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt	
+++ b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt	
@@ -122,6 +122,7 @@
             val shapes = StreamShapes.defaultShapes()
             val messageComposerTheme = MessageComposerTheme.defaultTheme(isInDarkMode, typography, shapes, colors)
             ChatTheme(
+                useDefaultSystemMediaPicker = true,
                 isInDarkMode = isInDarkMode,
                 colors = colors,
                 shapes = shapes,

```

</details>

Button scrollability:
1. Open the compose sample app on a smaller screen device
2. Ensure the system media picker factory is used (`useDefaultSystemMediaPicker`)
3. Ensure all 4 options from the attachments picker are shown
4. The last option should not be 'squashed', but rather the whole list should be scrollable.

#### General
- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [ ] PR targets the `develop` branch
- [ ] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerMediaCaptureTabFactory.kt [hunk 1]
```diff
@@ -120,36 +119,4 @@ public class AttachmentsPickerMediaCaptureTabFactory(private val pickerMediaMode
...
-    }
-
-    /**
-     * Define which media type will be allowed.
-     */
-    public enum class PickerMediaMode {
-        PHOTO,
-        VIDEO,
-        PHOTO_AND_VIDEO,
-    }
```
Comment (kanat) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerMediaCaptureTabFactory.kt:144]: Hey @VelikovPetar 👋 

We can't delete public classes within a single PR due to compatibility reasons.
We have to deprecate this one first with `@Deprecated`  annotation:
-  **warning**  level
- **error** level
- then delete 🙂

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerSystemTabFactory.kt [hunk 2]
```diff
@@ -56,23 +61,45 @@ import androidx.compose.ui.unit.dp
...
  */
-public class AttachmentsPickerSystemTabFactory(private val otherFactories: List<AttachmentsPickerTabFactory>) :
-    AttachmentsPickerTabFactory {
+public class AttachmentsPickerSystemTabFactory(
+    private val filesAllowed: Boolean,
+    private val mediaAllowed: Boolean,
+    private val captureImageAllowed: Boolean,
+    private val captureVideoAllowed: Boolean,
+    private val pollAllowed: Boolean,
+) : AttachmentsPickerTabFactory {
```
Comment (kanat) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerSystemTabFactory.kt:96]: This seems to be a braking change as well. Would be nice leave backward compatibility and maybe mark it as `@Deprecated` 👍

**Comment by kanat** [comment]
@VelikovPetar could you please add the ticket number to the PR title like `[PBE-XXX] <decription>` and the ticket link to the PR description? 🙂

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerMediaCaptureTabFactory.kt (hunk 1 above)
Comment (VelikovPetar) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerMediaCaptureTabFactory.kt]: Addressed here: 6dea00f188f482d882fbe8e4a89f05c574c08dd9

I basically reverted this change, as I realised there wasn't really a need to extract this class.

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerSystemTabFactory.kt (hunk 2 above)
Comment (VelikovPetar) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerSystemTabFactory.kt:98]: Addressed here: 20746a0e3ce1f61fcc8419ffd601ac1fcb669145

The current constructor is left as it was, but is now deprecated in favour of the newly added one. (Same goes for the `AttachmentsPickerTabFactories` default methods.)

**Comment by VelikovPetar** [comment]
> @VelikovPetar could you please add the ticket number to the PR title like `[PBE-XXX] <decription>` and the ticket link to the PR description? 🙂

I added the missing info, I think its more clear now 🙂

**Review by kanat — APPROVED** [review]
LGTM

**Review by aleksandar-apostolov — APPROVED** [review]

**Pull request merged**